src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceWrapper.cs

```diff
@@ -1,29 +1,44 @@
 ﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
 // Licensed under the MIT License.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
+using Microsoft.AspNet.OData.Common;
 using Microsoft.OData;
 
 namespace Microsoft.AspNet.OData.Formatter.Deserialization
 {
     /// <summary>
     /// Encapsulates an <see cref="ODataResource"/> and the inner nested resource infos.
     /// </summary>
-    public sealed class ODataResourceWrapper : ODataItemBase
-    {
+    public class ODataResourceWrapper : ODataItemBase
+    {       
         /// <summary>
         /// Initializes a new instance of <see cref="ODataResourceWrapper"/>.
         /// </summary>
         /// <param name="item">The wrapped item.</param>
-        public ODataResourceWrapper(ODataResource item)
+        public ODataResourceWrapper(ODataResourceBase item)
             : base(item)
         {
             NestedResourceInfos = new List<ODataNestedResourceInfoWrapper>();
         }
 
+  
+        /// <summary>
+        /// Gets the wrapped <see cref="ODataResourceBase"/>.
+        /// </summary>
+        public ODataResourceBase ResourceBase
+        {
+            get
+            {
+                return Item as ODataResourceBase;             
+            }
+        }
```

Comment:
I'd like to change this property as:

public ODataResourceBase ResourceBase {get;}

in the ctor,  public ODataResourceWrapper(ODataResourceBase item)
            : base(item)
        {
            ResourceBase = item;
            ...
        }

Reply:
+1

---
In reply to: [512293497](https://github.com/OData/WebApi/pull/2275#discussion_r512293497) [](ancestors = 512293497)